qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -131,9 +138,15 @@ def run(self, dag):
                 continue
 
             if dag.calibrations and node.name in dag.calibrations:
-                qubit = tuple([node.qargs[0].index])
-                params = tuple(node.op.params)
-                if (qubit, params) in dag.calibrations[node.name]:
+                qubits = tuple(qubit.index for qubit in node.qargs)
+                params = []
+                for p in node.op.params:
+                    if isinstance(p, ParameterExpression) and not p.parameters:
+                        params.append(float(p))
+                    else:
+                        params.append(p)
+                params = tuple(params)
+                if (qubits, params) in dag.calibrations[node.name]:
```

Comment:
Can this check be abstracted into its own function (for readability)? This pass is already a bit of an eyeful :).

Reply:
yes great suggestion 👍 👍

Reply:
I've added to basis_translator, it could be used in unroller too. What would you do? a utils file for this seems too nonlocal

Reply:
Agree, maybe as a method on the `DAGCircuit`?